frontend/lib/main.ts

```diff
@@ -58,7 +58,16 @@ window.addEventListener("load", () => {
   startApp(div, initialProps);
   polyfillSmoothScroll();
   showSafeModeUiOnShake();
-});
+}
+
+if (
+  document.readyState === "interactive" ||
+  document.readyState === "complete"
+) {
+  init();
+} else {
+  window.addEventListener("DOMContentLoaded", init);
```

Comment:
It's quite possible this will never actually be executed in practice, but I wanted to add it just in case.  It also ensures the script will work regardless of whether its script tag has the `async` attribute or not.